Remotion/Web/Development.Analyzers/RMWEB0001_WebStringAnalyzer.cs

```diff
@@ -0,0 +1,223 @@
+
+            if (webStringTypeSymbol == null
+                || plainTextStringTypeSymbol == null
+                || stringBuilderTypeSymbol == null
+                || htmlTextWriterSymbol == null)
+            {
+              return;
+            }
+
+            var objectSymbol = compilationContext.Compilation.GetTypeByMetadataName ("System.Object");
+            var stringSymbol = compilationContext.Compilation.GetTypeByMetadataName ("System.String");
+            var webStringEnumerable = compilationContext.Compilation.GetTypeByMetadataName ("System.Collections.Generic.IEnumerable`1")!
+                .Construct (webStringTypeSymbol);
+            var plainTextStringEnumerable = compilationContext.Compilation.GetTypeByMetadataName ("System.Collections.Generic.IEnumerable`1")!
+                .Construct (plainTextStringTypeSymbol);
+            var stringBuilderAppendMethodSymbol = GetStringBuilderAppendObjectMethodSymbol (stringBuilderTypeSymbol, objectSymbol);
+            var htmlTextWriterWriteMethodSymbol = GetHtmlTextWriterWriteObjectMethodSymbol (htmlTextWriterSymbol, objectSymbol);
+            var stringJoinMethodSymbols = GetStringJoinMethodSymbols (stringSymbol!);
+
+            var symbolContext = new SymbolContext (
+                webStringTypeSymbol,
+                plainTextStringTypeSymbol,
+                webStringEnumerable,
+                plainTextStringEnumerable,
+                stringBuilderAppendMethodSymbol,
+                htmlTextWriterWriteMethodSymbol,
+                stringJoinMethodSymbols);
+
+            compilationContext.RegisterSyntaxNodeAction (
+                ctx => AnalyzeInvocationExpression (ctx, symbolContext),
+                ImmutableArray.Create (SyntaxKind.InvocationExpression));
+          });
+    }
+
+    private void AnalyzeInvocationExpression (SyntaxNodeAnalysisContext ctx, SymbolContext symbols)
+    {
+      var invocationSymbolInfo = ctx.SemanticModel.GetSymbolInfo (ctx.Node);
+
+      if (invocationSymbolInfo.Symbol == null)
+        return;
+
+      var node = (InvocationExpressionSyntax) ctx.Node;
+      var invocationOriginalDefinitionSymbol = invocationSymbolInfo.Symbol.OriginalDefinition;
+
+      if (SymbolEqualityComparer.Default.Equals (invocationOriginalDefinitionSymbol, symbols.StringBuilderAppendMethodSymbol))
+      {
+        var argumentType = GetFirstArgumentTypeSymbol (node, ctx.SemanticModel);
+        if (SymbolEqualityComparer.Default.Equals (argumentType, symbols.WebStringTypeSymbol)
+            || SymbolEqualityComparer.Default.Equals (argumentType, symbols.PlainTextStringTypeSymbol))
+        {
+          var diagnostic = Diagnostic.Create (
+              DiagnosticDescriptor,
+              ctx.Node.GetLocation(),
+              symbols.StringBuilderAppendMethodSymbol.ToString(),
+              argumentType.ToString(),
+              "Call '.ToString(WebStringEncoding.HtmlWithTransformedLineBreaks)' on the WebString instance instead.");
```

Comment:
Don't we need to cover WebString and PlainTextString separately? The message only refers to WebString.

Reply:
Whoops, forgot this occurrence after adding plaintextstring support